bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,473 @@
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.tibber.internal.config.TibberConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link TibberHandler} is responsible for handling queries to/from Tibber API.
+ *
+ * @author Stian Kjoglum - Initial contribution
+ */
+@NonNullByDefault
+public class TibberHandler extends BaseThingHandler {
+    private static final int REQUEST_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(20);
+    private final Logger logger = LoggerFactory.getLogger(TibberHandler.class);
+    private final Properties httpHeader = new Properties();
+    private final SslContextFactory sslContextFactory = new SslContextFactory(true);
+    private final Executor websocketExecutor = ThreadPoolManager.getPool("tibber.websocket");
+    private final TibberWebSocketListener socket = new TibberWebSocketListener();
+    private @Nullable Session session;
+    private @Nullable WebSocketClient client;
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private @NonNullByDefault({}) String myToken;
+    private @NonNullByDefault({}) String myHome;
+
+    private int refresh;
+    private boolean connected = false;
+    private String rtEnabled = "false";
+
+    public TibberHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        TibberConfiguration tibberConfig = getConfigAs(TibberConfiguration.class);
+        myToken = tibberConfig.getToken();
+        myHome = tibberConfig.getHomeid();
+        refresh = tibberConfig.getRefresh();
```

Comment:
It is standard in other bindings to store the config as a field and then retrieve configs from it when you need them. Please do the same, it would cleanup fields by using one config fields instead.

Reply:
I struggled with nullness and thus created local config parameters as a workaround. Have now managed to store config in one field and retrieving config parameters from it as needed.